File: Tier1/Shared/DTOs/AgreementListDTO.cs
using Shared.Domain;

namespace Shared.DTOs;

public class AgreementListDTO
{
    public List<Agreement> AgreementList { get; set; }
    public string ErrorMessage { get; set; }

    public AgreementListDTO(List<Agreement> agreementList, string errorMessage)
    {
        AgreementList = agreementList;
        ErrorMessage = errorMessage;
    }

    public AgreementListDTO()
    {
    }
}

File: Tier2/src/main/java/via/sep3/group11/tier2/shared/DTOs/AgreementListDTO.java
package via.sep3.group11.tier2.shared.DTOs;

import via.sep3.group11.tier2.shared.domain.Agreement;

import java.util.List;

public class AgreementListDTO
{
    private List<Agreement> agreementList;
    private String errorMessage;

    public AgreementListDTO(List<Agreement> agreementList, String errorMessage) {
        this.agreementList = agreementList;
        this.errorMessage = errorMessage;
    }

    public List<Agreement> getAgreementList() {
        return agreementList;
    }

    public String getErrorMessage() {
        return errorMessage;
    }
}
